.evergreen/install-composer.sh

```diff
@@ -38,4 +38,9 @@ php --ri mongodb
 
 install_composer
 
+# Remove psalm as it's not compatible with PHP 8.4: https://github.com/vimeo/psalm/pull/10928
+if [ "$PHP_VERSION" == "8.4" ]; then
+  php composer.phar remove --no-update --dev vimeo/psalm
```

Comment:
Since we only run PHPUnit, I decided to explicitly remove the psalm dependency instead of ignoring any upper bounds for PHP version constraints, as that may result in actually incompatible packages being pulled in.